Framework/Source/API/VariablesBuffer.h

```diff
@@ -93,18 +93,11 @@ namespace Falcor
         template<typename T>
         void setVariableArray(const std::string& name, size_t elementIndex, const T* pValue, size_t count);
 
-        void setTexture(const std::string& name, const Texture* pTexture, const Sampler* pSampler);
-
-        void setTextureArray(const std::string& name, const Texture* pTexture[], const Sampler* pSampler, size_t count);
-
-        void setTexture(size_t Offset, const Texture* pTexture, const Sampler* pSampler);
-
-        void setTextureInternal(size_t offset, const Texture* pTexture, const Sampler* pSampler);
-
-        ProgramReflection::BufferReflection::SharedConstPtr mpReflector;
+        ReflectionResourceType::SharedConstPtr mpReflector;
         std::vector<uint8_t> mData;
         mutable bool mDirty = true;
         size_t mElementCount;
         size_t mElementSize;
+        std::string mName;
```

Comment:
Rather than store both a name and a reflection type, couldn't this in principle just store a pointer to a reflection variable?

Reply:
The current interface forces the user to pass a `ReflectionResourceType` which can checked during compilation. I think it makes my intentions clearer (i.e. no confusion as to what var with array-of-resource-types mean)